doc/api/esm.md

```diff
@@ -0,0 +1,86 @@
+# Modules
```

Comment:
A link to this file should be added to `doc/api/_toc.md`.
Maybe change the title to "ECMAScript Modules"? There is already a page called "Modules".